server/server.go

```diff
@@ -603,6 +605,14 @@ func (s *Server) handlePublishWithoutResponse(r *http.Request, v *visitor) (*mes
 	if err != nil {
 		return nil, err
 	}
+	v_old := v
+	if strings.HasPrefix(t.ID, subscriberBilledTopicPrefix) {
+		v = t.getBillee()
+		if v == nil {
+			return nil, errHTTPWontStoreMessage
+		}
+	}
```

Comment:
I did this using context (as suggested below), but isn't that supposed to be an anti-pattern? In any case, I guess it's just one thing so it shouldn't matter much. I feel like the ideal solution to this would be making `visitor` an interface and having a `compositeVisitor` in addition to visitor...but, that's probably overengineering.